OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/purity/L1PurityAnalysis.scala

```diff
@@ -278,12 +282,14 @@ class L1PurityAnalysis private[analyses] (val project: SomeProject) extends Abst
                     return Result(state.context, ImpureByAnalysis)
 
             // Cases that are pure
-            case FinalP(_: FinalField)                   => // Reading eff. final fields
-            case FinalP(ImmutableType | ImmutableObject) => // Returning immutable reference
+            case FinalP(NonAssignable | EffectivelyNonAssignable | LazilyInitialized) => // Reading eff. final fields
+            case FinalP(TransitivelyImmutableType |
+                TransitivelyImmutableClass) => // Returning immutable reference
 
             // Cases resulting in side-effect freeness
-            case FinalP(_: FieldMutability | // Reading non-final field
-                _: TypeImmutability | _: ClassImmutability) => // Returning mutable reference
+            case FinalP(_: FieldAssignability | // Reading non-final field
+                _: TypeImmutability |
```

Comment:
Probably not necessary to put everything on a line of its own, especially since the comment referens to both, class and type immutability